openapi/types.rs

```diff
@@ -0,0 +1,66 @@
+use ordered_float::OrderedFloat;
+use serde::{Deserialize, Serialize};
+use utoipa::{IntoParams, ToSchema};
+use uuid::Uuid;
+
+use crate::structs::{AssetStatus, Location};
+
+/// Request to create an aircraft.
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, IntoParams)]
+pub struct RegisterAircraftPayload {
+    pub name: Option<String>,
+    /// The UUID of an [`AssetGroup`] struct, if available.
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub whitelist: Vec<Uuid>,
+    pub status: AssetStatus,
+    /// The aircraft's manufacturer.
+    ///
+    /// TODO: For now we can just say "Boeing", "Airbus", etc. Later, we
+    /// can a struct for this and store the manufacturer's name, logo,
+    /// etc.
+    pub manufacturer: String,
+    pub model: String,
+    pub serial_number: String,
+    /// The aircraft's registration number.
+    ///
+    /// In the US, this is the N number.
+    ///
+    /// This is a unique identifier for
+    /// the aircraft that can be used to look up information about the
+    /// aircraft from national aviation authorities like the FAA.
+    pub registration_number: String,
+    pub description: Option<String>,
+    pub max_payload_kg: OrderedFloat<f32>,
+    pub max_range_km: Option<OrderedFloat<f64>>,
+}
+
+/// Request to create a vertiport.
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, IntoParams)]
+pub struct RegisterVertiportPayload {
+    pub name: Option<String>,
+    /// The UUID of an [`AssetGroup`] struct, if available.
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub whitelist: Vec<Uuid>,
+    pub status: AssetStatus,
+    pub description: Option<String>,
+    pub location: Location,
+    pub vertipads: Vec<Uuid>,
+}
+
+/// Request to create a vertipad.
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, IntoParams)]
+pub struct RegisterVertipadPayload {
```

Comment:
We already have a GRPC struct for the vertipads, we should align the REST requests with those fields (missing fields, and what does the asset status describe?):
https://github.com/Arrow-air/svc-storage/blob/main/client-grpc/src/grpc.vertipad.rs#L13

Reply:
Status indicates whether this vertipad is available for use. So, if there's some maintenance going on, the status would be unavailable.

Reply:
Will use storage types in another PR.